flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/control/ProcessVariableSupplierPidController.java

```diff
@@ -0,0 +1,36 @@
+package com.flash3388.flashlib.robot.control;
+
+import com.flash3388.flashlib.time.Clock;
+import com.jmath.ExtendedMath;
+
+import java.util.function.DoubleSupplier;
+import java.util.function.DoubleUnaryOperator;
+
+public class ProcessVariableSupplierPidController extends PidController implements DoubleUnaryOperator {
```

Comment:
Name tooo long. Plus this heirarchy is a bit weird, since it exposes two conflict APIs, one where you use `applyAsDouble(setpoint)`, and one where you use `calculate(processVariable, setpoint, time)`. 

Try looking for a different inheritance organization. Maybe some interface or abstract class.

Reply:
I didn't like the naming too, can change the `calculate()` to `applyAsDouble()` if that would help... Don't have many good ideas about that tn

Reply:
You misunderstood. Someone using `ProcessVariableSupplierPidController` can also use the original `calculate(processVariable, setpoint, time)`, which completely bypasses the supplier here. This is unwanted, and thus the API would have to be changed.

Reply:
It was intended, like, in case he wanted to supply this time himself

Reply:
but I guess it's also problematic

Reply:
So... may be just receive the basic Pid controller instead of extending it?